lib/ops/events/events.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"github.com/gravitational/gravity/lib/ops"
+
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+var log = logrus.WithField(trace.Component, "events")
+
+// Emit saves the provided event to the audit log of the local cluster of the
+// provided operator.
+func Emit(operator ops.Operator, event string, fields Fields) {
+	err := emit(operator, event, fields)
+	if err != nil {
+		log.Errorf("Failed to emit audit event %v %v: %v.",
+			event, fields, trace.DebugReport(err))
+	}
+}
+
+func emit(operator ops.Operator, event string, fields Fields) error {
+	cluster, err := operator.GetLocalSite()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return operator.EmitAuditEvent(ops.AuditEventRequest{
+		SiteKey: cluster.Key(),
+		Type:    event,
+		Fields:  events.EventFields(fields),
+	})
+}
+
+const (
+	// OperationStarted is emitted when an operation starts.
+	OperationStarted = "operation.started"
+	// OperationCompleted is emitted when an operation completes successfully.
+	OperationCompleted = "operation.completed"
+	// OperationFailed is emitted when an operation completes with error.
+	OperationFailed = "operation.failed"
+
+	// AppInstalled is emitted when an application image is installed.
+	AppInstalled = "application.installed"
+	// AppUpgraded is emitted when an application release is upgraded.
+	AppUpgraded = "application.upgraded"
+	// AppRolledBack is emitted when an application release is rolled back.
+	AppRolledBack = "application.rolledback"
+	// AppUninstalled is emitted when an application release is uninstalled.
+	AppUninstalled = "application.uninstalled"
+
+	// ResourceCreated is emitted when a Gravity resource is created or updated.
+	ResourceCreated = "resource.created"
```

Comment:
nit: `resource.updated` would be more appropriate

Reply:
I was going for an analogy with "gravity resource" subcommands. I think ideally we'd have both created and updated for that's for later (which likely means "never" if we're honest :) )